Replace the experimental/prototype-wasmate/test tests.

sunfishcode opened the pull request:
Use the llvm_autogenerated tests instead.

kripken commented on .gitmodules:
```diff
@@ -1,6 +1,6 @@
 	url = https://github.com/WebAssembly/testsuite.git
-[submodule "test/experimental"]
-	path = test/experimental
+[submodule "experimental"]
```
why is there still a submodule?

sunfishcode replied:
I had initially removed it, but check.py does `import experimental.buildbot.link_assembly_files as link_assembly_files` and needs it.

jfbastien replied:
I'll move that to https://github.com/WebAssembly/waterfall soon, I just need a mirror in the buildbots to prevent flakes (waiting on Chrome infra to do this).

kripken commented on README.md:
```diff
@@ -171,7 +171,7 @@ The `check.py` script supports some options:
  * If tests are provided, we run exactly those. If none are provided, we run them all.
  * Some tests require `emcc` or `nodejs` in the path. They will not run if the tool cannot be found, and you'll see a warning.
- * We have tests from upstream in `tests/spec` and `tests/experimental`, in git submodules. Running `./update.py` should update those.
+ * We have tests from upstream in `tests/spec` in git submodules. Running `./update.py` should update those.
```
no longer plural in both sentences

sunfishcode replied:
Fixed.

kripken commented on check.py:
```diff
@@ -191,26 +191,6 @@ def fix(x):
     if actual != expected:
       fail(actual, expected)
 
-print '\n[ checking binaryen-shell experimental testcases... ]\n'
```
we should still run on those wasts, in their new location?

kripken replied:
or does that happen lower down and i missed it?

sunfishcode replied:
These are the tests that are now in llvm_autogenerated, which is added to the script along with the dot_s tests.

kripken replied:
Yes, and I see that `s2wasm` is run on those. However, we previously also ran `binaryen-shell` on the wasts, to verify that it could parse them, which your code removed.

This isn't what the tests were intended for, I suppose, but I did notice that those wasts had a bunch of stuff not in the spec wasts, so it was useful to verify binaryen can parse them.

sunfishcode replied:
Would it make sense to run binaryen-shell on all the dot_s tests too?

kripken replied:
On their wast expected outputs, yes.

sunfishcode commented:
Updated:
 - run binaryen-shell on the expected .wast files for all dot_s and llvm_autogenerated tests
 - fixed a bug in hex literal printing found in immediates.s
 - remove .wast files from dot_s that are now in llvm_autogenerated

kripken commented:
lgtm, assuming `check.py` passes and @jfbastien.

kripken commented:
`check.py` looks good, so just waiting on @jfbastien for feedback. Looks like there is also a merge conflict now though.

jfbastien commented:
I think the conflict is just on the submodule moving out of test/, which doesn't seem necessary.

Rest LGTM, I'll update it with the waterfall infra once committed.

kripken commented:
Perhaps the submodule move is necessary for a later step @sunfishcode has in mind?

In any case, @sunfishcode feel free to resolve and merge.

sunfishcode commented:
Ok, I reverted the submodule change. It wasn't actually important for anything; I had just removed it and then later re-added it and didn't happen to re-add it in the same place. It's now back to test/experimental.

kripken commented:
Do we still need that experimental submodule for anything, @sunfishcode and @jfbastien ?

sunfishcode commented:
There's still the code in check.py that does `import test.experimental.buildbot.link_assembly_files as link_assembly_files` and so on which is using it.

kripken commented:
Right, ok.

jfbastien commented:
Yeah, I'll move it once the code lives in https://github.com/WebAssembly/waterfall. I'm waiting for the bots to mirror the github repo first (otherwise we DDoS github.com, or builds flake when github.com is down).
